source/tests/unit/ni_fake_non_ivi_service_tests.cpp

```diff
@@ -1508,6 +1514,36 @@ TEST_F(NiFakeNonIviServiceTests, InitFromCrossDriverSession)
   EXPECT_EQ(kDriverSuccess, response.status());
 }
 
+template <size_t N>
+InitFromCrossDriverSessionArrayRequest create_init_from_cross_driver_session_array_request(const std::array<std::string, N>& session_names)
+{
+  InitFromCrossDriverSessionArrayRequest request;
+  request.set_session_name("fake_session");
+  for (const auto& session_name : session_names) {
+    auto session_in = request.add_cross_driver_session_array();
+    session_in->set_name(session_name);
+  }
+  return request;
+}
+
+TEST_F(NiFakeNonIviServiceTests, InitFromCrossDriverSessionArray_PassesSessionResourcesAndSucceeds)
+{
+  constexpr auto CROSS_DRIVER_HANDLES = std::array<FakeCrossDriverHandle, 2>{1234, 5678};
+  const auto CROSS_DRIVER_SESSION_NAMES = std::array<std::string, 2>{"cross_driver_session", "cross_driver_session_2"};
+  add_session(cross_driver_resource_repository_, CROSS_DRIVER_HANDLES[0], CROSS_DRIVER_SESSION_NAMES[0]);
+  add_session(cross_driver_resource_repository_, CROSS_DRIVER_HANDLES[1], CROSS_DRIVER_SESSION_NAMES[1]);
+  EXPECT_CALL(library_, InitFromCrossDriverSessionArray(_, _, _))
+      .With(Args<0, 1>(ElementsAreArray(CROSS_DRIVER_HANDLES.data(), CROSS_DRIVER_HANDLES.size())))
+      .WillOnce(DoAll(
+          SetArgPointee<2, FakeHandle>(4567), Return(kDriverSuccess)));
```

Comment:
Question: Why 4567?

Reply:
Can you check out [this commit](https://github.com/ni/grpc-device/pull/479/commits/41ba1b8975b062f4d33f7f114a4b81233358407f) and see if it helps explain it? 

This is the internal driver session (not the cross-driver session). For this test, it just needs to be a non-zero integer so that there's something to add to the map. If we were to call other methods with the grpc session returned by init, it would resolve to that session and pass it to the driver C API (or mock library in this case). 

For this test, we're more concerned that the right stuff gets passed into the init method (the EXPECT above) and what happens after that is covered more in other tests. But ... I agree its confusing to have a magic number, so I tried adding some asserts/constants that explain it.